app/scripts/background.js

```diff
@@ -919,12 +918,11 @@ async function onInstall() {
   const storeAlreadyExisted = Boolean(await localStore.get());
   // If the store doesn't exist, then this is the first time running this script,
   // and is therefore an install
-  if (
-    !storeAlreadyExisted &&
-    !(process.env.METAMASK_DEBUG || process.env.IN_TEST)
-  ) {
+  if (!storeAlreadyExisted && !process.env.METAMASK_DEBUG) {
     addAppInstalledEvent();
-    platform.openExtensionInBrowser();
+    if (!process.env.IN_TEST) {
+      platform.openExtensionInBrowser();
+    }
   }
```

Comment:
Explanation of condition change:
* When we are in testing mode, we still want the appInstall event to be sent, otherwise we cannot test this event.
* Condition for the other action `platform.openExtensionInBrowser()` stays the same: we only execute it when we are not in debug or test mode, and when  it's the first time to install application.